### ISSUE TITLE
Disable RT-kernel sanity check for nvidia-x11

### ISSUE THREAD

**Issue by fps** [musnix/musnix#127]
Hi,

to get the nividia driver module to build against the RT kernels I needed to add this diff:

<pre>
index 5baa4aa51e5..70fc5126a93 100755
--- a/pkgs/os-specific/linux/nvidia-x11/builder.sh
+++ b/pkgs/os-specific/linux/nvidia-x11/builder.sh
@@ -20,7 +20,7 @@ buildPhase() {
         sysSrc=$(echo $kernel/lib/modules/$kernelVersion/source)
         sysOut=$(echo $kernel/lib/modules/$kernelVersion/build)
         unset src # used by the nv makefile
-        make SYSSRC=$sysSrc SYSOUT=$sysOut module -j$NIX_BUILD_CORES
+        make IGNORE_PREEMPT_RT_PRESENCE=1 SYSSRC=$sysSrc SYSOUT=$sysOut module -j$NIX_BUILD_CORES
 
         cd ..
     fi
</pre>

I'm not sure how to best proceed to make this available to other users. An option for the nvidia-x11 derivation? Or some hackery in musnix? I'm open to your input..

**Comment by pushqrdx** [comment]
@fps can you please help me, i also want to apply this patch but idk how? a simple overlay doesn't work

**Comment by fps** [comment]
Here is how I did it:

1. Clone nixpkgs
2. Optionally create a branch first <code>git checkout -b rt-remove-sanity-check</code>
3. Apply this patch (usually something like <code>diff -Nru --dry-run -p1 < the.diff</code> to see if it applies cleanly and then <code>diff -Nru -p1 < the.diff</code> to really do it. Both from the nixpkgs root directory).
4. Rebuild your system <code>nixos-rebuild switch -I nixpkgs=/path/to/your/nixpkgs</code>

**Comment by fps** [comment]
If you don't want to patch nixpkgs I can imagine that you could create an overlay with a package called <code>nvidia-x11-rt</code> and use that in your system instead of <code>nvidia-x11</code>. But I have no experience with overlays..

**Comment by pushqrdx** [comment]
Yeah i was insisting on doing it with an overlay which turned out virtually impossible because nvidia-x11 has a dependency on the kernel. So i went with a clone of nixpkgs and it worked. Anyway thanks for your help!
